components/board/fake/board.go

```diff
@@ -25,7 +25,7 @@ import (
 type Config struct {
 	I2Cs              []board.I2CConfig              `json:"i2cs,omitempty"`
 	SPIs              []board.SPIConfig              `json:"spis,omitempty"`
-	Analogs           []board.AnalogConfig           `json:"analogs,omitempty"`
+	AnalogReaders     []board.AnalogReaderConfig     `json:"analog_readers,omitempty"`
```

Comment:
note that this is breaking change for fake board config, is this fine? Could just keep json as `analogs` if we don't want to break

Reply:
I like changing the name eventually because if we keep it as "analogs" and later add "analog_writers" it'll be super confusing. However, there are hundreds of pre-existing robot configs using a fake board with a non-empty "analogs". I think we should rename them all. This likely involves creating a migration script similar to [this example](https://github.com/viamrobotics/app/tree/main/etc/migrations/fake_motor_config_dir_flip) (which renamed "direction_flip" to "dir_flip" on the fake motor). 

We could split the difference and keep the json name as "analogs" for now, and change the name when analog writers come in later. That way, we don't have a breaking change yet, which gives us more time to think about whether there's a different way to do this before we impact users.

Reply:
Since the analog write function in being called on the board I do not think we will need an `AnalogWriter` config in the fake board. So I will change back to analogs.

Reply:
Actually even with changing to `analogs ` it would still be breaking the fake board config, removing `SpiBus` and `ChipSelect` keys. Would calling this out in breaking changes be enough or do I need to do a migration to remove these attributes from the db?

Reply:
Hm... not sure. Having extra items in the config that no longer do anything won't _hurt_ anything, it'll just look weird/redundant. So, maybe just calling it out on Slack is enough. but if you leave redundant/useless things laying around, they get copied and pasted and propagated, and people assume that because they're there they must do _something,_ leading to confusion. So, the cleanest thing to do is to write a migration script, but that's a lot of effort for something that doesn't change how the code works (it just changes how the humans interpret the configs). Is that worth it? I'm unsure; get another couple opinions.